modules/farm_fd2/src/entrypoints/cover_crop_seeding/lib.submitOptional.unit.cy.js

```diff
@@ -1,121 +1,142 @@
 import { lib } from './lib.js';
+import * as farmosUtil from '@libs/farmosUtil/farmosUtil';
 
-describe('Submission using the cover_crop lib.', () => {
-  beforeEach(() => {
-    cy.restoreLocalStorage();
-    cy.restoreSessionStorage();
-  });
+describe('Submission scenarios for cover crop lib', () => {
+  let equipmentMap;
 
-  afterEach(() => {
-    cy.saveLocalStorage();
-    cy.saveSessionStorage();
+  before(() => {
+    const timeout = { timeout: 30000 };
+    cy.wrap(farmosUtil.getEquipmentNameToAssetMap(), timeout).then((map) => {
+      equipmentMap = map;
+    });
   });
 
-  it('Check that winter kill log is not created if winterKill is false', () => {
-    cy.wrap(
-      lib.submitForm({
-        date: '1950-01-02',
-        crops: ['BEAN', 'CARROT'],
-        location: 'ALF',
-        beds: ['ALF-1', 'ALF-3'],
-        areaSeeded: 50,
-        seedApplicationEquipment: ['Tractor'],
-        seedIncorporationEquipment: ['Rake'],
-        seedApplicationDepth: 6,
-        seedApplicationSpeed: 5,
-        seedIncorporationDepth: 8,
-        seedIncorporationSpeed: 3,
-        winterKill: false,
-        comment: 'A comment',
-      }),
-      { timeout: 10000 }
-    ).then((res) => {
-      expect(res.plantAsset).to.exist;
-      expect(res.areaSeededQuantity).to.exist;
-      expect(res.seedingLog).to.exist;
-      expect(res.winterKillLog).to.be.null;
-      expect(res.seedApplicationDepthQuantity).to.exist;
-      expect(res.seedApplicationSpeedQuantity).to.exist;
-      expect(res.seedApplicationAreaQuantity).to.exist;
-      expect(res.seedApplicationActivityLog).to.exist;
-      expect(res.seedIncorporationDepthQuantity).to.exist;
-      expect(res.seedIncorporationSpeedQuantity).to.exist;
-      expect(res.seedIncorporationAreaQuantity).to.exist;
-      expect(res.seedIncorporationActivityLog).to.exist;
+  // No winter kill
+  describe('When winter kill is false', () => {
+    let results = null;
+    const form = {
+      date: '1950-01-02',
+      crops: ['BEAN'],
+      location: 'ALF',
+      beds: [],
+      areaSeeded: 50,
+      seedApplicationEquipment: ['Tractor'],
+      seedIncorporationEquipment: ['Rake'],
+      seedApplicationDepth: 6,
+      seedApplicationSpeed: 5,
+      seedApplicationPasses: 1,
+      seedIncorporationDepth: 8,
+      seedIncorporationSpeed: 3,
+      seedIncorporationPasses: 1,
+      winterKill: false,
+      winterKillDate: null,
+      comment: 'A comment',
+    };
+
+    before(() => {
+      cy.wrap(lib.submitForm(form), { timeout: 30000 }).then((res) => {
+        results = res;
+      });
+    });
+
+    it('should create all standard logs', () => {
+      expect(results).to.have.property('plantAsset');
+      expect(results).to.have.property('seedingLog');
+      expect(results).to.have.property('seedApplicationActivityLog0');
+    });
+
```

Comment:
It would be a good idea to check for the quantities as well.  I think that all of the `expects' in this describe, including new ones for the quantities can be contained in a single `it`.